x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/field_stats_flyout/field_stats_flyout.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, {
+  createContext,
+  FC,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { FieldStats, FieldStatsServices } from '@kbn/unified-field-list-plugin/public';
+import moment from 'moment';
+import {
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiFlyoutBody,
+  useGeneratedHtmlId,
+  EuiFlyoutFooter,
+  EuiButton,
+  EuiButtonIcon,
+  EuiToolTip,
+  EuiSpacer,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { css } from '@emotion/react';
+import { getDefaultDatafeedQuery } from '../../../../../utils/new_job_utils';
+import { useMlKibana } from '../../../../../../../contexts/kibana';
+import { JobCreatorContext } from '../../../job_creator_context';
+import { SingleMetricJobCreator } from '../../../../../common';
+import { isDefined } from '../../../../../../../../../common/types/guards';
+
+const defaultDatafeedQuery = getDefaultDatafeedQuery();
+
+export const FieldStatsContent: FC = () => {
+  const {
+    services: { uiSettings, data, fieldFormats, charts },
+  } = useMlKibana();
+  const fieldStatsServices: FieldStatsServices = {
+    uiSettings,
+    dataViews: data.dataViews,
+    data,
+    fieldFormats,
+    charts,
+  };
+  const { jobCreator: jc, jobCreatorUpdated } = useContext(JobCreatorContext);
+  const { fieldName } = useContext(MLJobWizardFieldStatsFlyoutContext);
+
+  const jobCreator = jc as SingleMetricJobCreator;
+
+  const [start, setStart] = useState(jobCreator.start);
+  const [end, setEnd] = useState(jobCreator.end);
+
+  useEffect(() => {
+    if (jobCreator.start !== start || jobCreator.end !== end) {
+      setStart(jobCreator.start);
+      setEnd(jobCreator.end);
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [jobCreatorUpdated]);
+
+  // Format timestamp to ISO formatted date strings
+  const timeRange = useMemo(
+    () =>
+      start && end
+        ? { from: moment(start).toISOString(), to: moment(end).toISOString() }
+        : undefined,
+    [start, end]
+  );
+
+  const fieldForStats = useMemo(
+    () => (isDefined(fieldName) ? jobCreator.dataView.getFieldByName(fieldName) : undefined),
+    [fieldName, jobCreator.dataView]
+  );
+
+  const showFieldStats = timeRange && isDefined(jobCreator.dataViewId) && fieldForStats;
+
+  return showFieldStats ? (
+    <FieldStats
```

Comment:
We could also try another approach by defining `key` prop. It would recreate the whole component if `key` changes which would include refetching.
```
<FieldStats 
  key={field.name}
  ...
/>
```

Reply:
Updated here [`2ab4d99` (#147322)](https://github.com/elastic/kibana/pull/147322/commits/2ab4d991e26d302b6865b78967aef8d060957c48)